### DISCUSSION TITLE
`ConvexSpheropolyhedronUnion` sets sweep_radius to 0

### DISCUSSION THREAD

**Discussion by janbridley** [glotzerlab/hoomd-blue#1875] [General]
https://github.com/glotzerlab/hoomd-blue/blob/5a19a7962de1642f622e46725fb0ef1b4987dde2/hoomd/hpmc/integrate.py#L1974

It appears that the `ConvexSpheropolyhedronUnion` class sets the sweep radius of union members to 0, with no interface to change this default. It seems logical that a radius should be a valid parameter - is this a functional limitation or an accidental omission? The value is not set in the ShapeUnion header file either, so either case seems possible.

https://github.com/glotzerlab/hoomd-blue/blob/5a19a7962de1642f622e46725fb0ef1b4987dde2/hoomd/hpmc/ShapeUnion.h#L139

**Comment by joaander** [comment]
`ShapeUnion` is templated on the `Shape` class. `ShapeUnion<ShapeSpheropolyhedron>` will therefore use `ShapeSpheropolyhedron`'s `param_type` constructor which reads `sweep_radius` from the given shape dictionary:
https://github.com/glotzerlab/hoomd-blue/blob/5a19a7962de1642f622e46725fb0ef1b4987dde2/hoomd/hpmc/ShapeUnion.h#L204-L216

For example:
```
mc.shape["A"] = dict(shapes=[{'vertices': cube_verts, 'sweep_radius': 0.5},
                             {'vertices': tetrahedron_verts, 'sweep_radius': 0.5}],
                     positions=[(0, 0, 0), (0, 0, 1)])
```

**Reply by janbridley** [reply]
Got it - so instead of just the vertices, we could pass in an entire shape spec for the rounded shape. Would it be helpful to update the docs to reflect this?

#### Previous docstring
<img width="801" alt="image" src="https://github.com/user-attachments/assets/c0ab39c4-8fba-4a08-9e9f-4e7d36db763f">

#### Proposed replacement

```python
        mc = hoomd.hpmc.integrate.ConvexSpheropolyhedronUnion(default_d=0.3,
                                                              default_a=0.4)
        cube = [[-1,-1,-1],
                [-1,-1,1],
                [-1,1,1],
                [-1,1,-1],
                [1,-1,-1],
                [1,-1,1],
                [1,1,1],
                [1,1,-1]]
        mc.shape["A"] = dict(shapes=[{'vertices': cube_verts, 'sweep_radius': 0.5},
                            		 {'vertices': cube_verts, 'sweep_radius': 0.0}],
                             positions=[(0, 0, 0), (0, 0, 1)],
                             orientations=[(1, 0, 0, 0), (1, 0, 0, 0)],
                             overlap=[1, 1]);
        print('vertices of the first cube = ',
              mc.shape["A"]["shapes"][0]["vertices"])
        print('center of the first cube = ', mc.shape["A"]["positions"][0])
        print('orientation of the first cube = ',
              mc.shape_param["A"]["orientations"][0])
```

**Reply by joaander** [reply]
I would remove of the `overlap=[1,1]` line as this is not a commonly used feature. Also remove the calls to `print` and wrap the whole block in a `.. code-block:: python` to test the code with Sybil.

In general it appears that most of the example blocks in `hpmc/integrate.py` are not yet tested with Sybil.
